esp/platform/espp.cpp

```diff
@@ -452,7 +452,18 @@ int init_main(int argc, const char* argv[])
         else
         {
             envpt.setown(createPTreeFromXMLFile(cfgfile, ipt_caseInsensitive));
+
+            // NB: esp has no standard component config in bare-metal, this is loading defaultYaml only
             espConfig.setown(loadConfiguration(defaultYaml, argv, "esp", "ESP", nullptr, nullptr));
```

Comment:
Curious again. Why not pass the cfgfile to the loadConfiguration like other components (for example, dfuserver: globals.setown(loadConfiguration(defaultYaml, argv, "dfuserver", "DFUSERVER", "dfuserver.xml", nullptr)))?

Reply:
it could, but it wouldn't be in the standard format, because the config that would be instantiated in component config is very different from the getComponentConfig format ubiquitous elsewhere
i.e. esp.xml is effectively a selective copy of the environment tree.
And, nothing in the esp code base relies on getComponentConfig* in esp, and can't do so as it stands, because before and after this change it is empty (well populated with defaultYaml only).
It could instantiate the config to the esp.xml config, but generic helper code that expects getComponentConfig to follow certain norms, wouldn't work.

I don't know there are any places in bare-metal in esp that rely on that though..

It is related to when cloud config came along, TonyF, refactored the config loading code in esp, so that at runtime, it builds a <Environment> based IPT from the esp specific 'esp:' section (that's what buildApplicationLegacyConfig is doing).
I think this was to avoid having to change much of the esp code to use the new config scheme.

Ideally, it would have been the other way around, so that legacy esp was translated into a standarized form that adhered to values.schema.json, and all esp code used it and getComponentConfig consistently - that way generic code elsewhere would also make sense in esp.
Definitely beyond the scope of this PR though.

For the most part, I think as long as bare-metal config options are consistently made available via getGlobalConfig, it's not a significant issue.